layer/layer_data.cc

```diff
@@ -72,15 +103,24 @@ LayerData::LayerData(char* log_filename, const char* header) {
   if (log_filename) {
     out_ = fopen(log_filename, "w");
     if (out_ == nullptr) {
-      fprintf(stderr, "Failed to open %s, output will be to STDERR.\n",
-              log_filename);
+      LOG(ERROR) << "Failed to open " << log_filename
+                 << ", output will be to STDERR.\n";
       out_ = stderr;
     }
   } else {
     out_ = stderr;
   }
-  fprintf(out_, "%s\n", header);
-  fflush(out_);
+
+  WriteLnAndFlush(out_, header);
+
+  if (const char* event_log_file = getenv(kEventLogFileEnvVar)) {
+    absl::MutexLock lock(&event_log_lock_);
```

Comment:
no need to take this lock in the constructor since no other threads will be able to access this object until after the constructor completes